src/libs/Launcher.js

```diff
@@ -368,6 +370,52 @@ export default class Launcher {
     return result
   }
 
+  /**
+   * Creates a trigger to launch a service from home application which will force the end of the current job with 'context deadline exceeded'
+   * message after 20 minutes
+   *
+   * @returns {Promise<void>}
+   */
+  async createTimeoutTrigger() {
+    const { client, job } = this.getStartContext() || {}
+
+    if (!job) {
+      throw new Error('createTimeoutTrigger: no job found in context')
+    }
+
+    const { data: timeoutTrigger } = await client.save({
+      _type: 'io.cozy.triggers',
+      worker: 'service',
+      type: '@at', // did not use @in because only @at triggers remove themselves after use
+      arguments: new Date(
+        Date.now() + constants.serviceTimeoutDuration
+      ).toISOString(),
+      message: {
+        slug: 'home',
+        name: 'cliskTimeout',
+        fields: {
+          cliskJobId: job._id
+        }
+      }
+    })
+    this.setStartContext({
+      ...this.getStartContext(),
+      timeoutTrigger
+    })
+  }
+
+  /**
+   * Removes the timeout trigger when the job is already done
+   *
+   * @returns {Promise<void>}
+   */
+  async removeTimeoutTrigger() {
+    const { client, timeoutTrigger } = this.getStartContext() || {}
+    if (timeoutTrigger) {
+      await client.destroy(timeoutTrigger)
```

Comment:
Can this throw if `timeoutTrigger` is not found on cozy-stack?

Reply:
Yes it could. I added try catch arount it